docs/develop/api-reference/auth-tokens.md

```diff
@@ -0,0 +1,174 @@
+| permissions           | Array <[Permission](#permission-objects)\> | The permissions to grant to the new token.|
+
+A TokenScope is an array of [permission objects](#permission-objects). The array can have permissions that are of type [CachePermission](#cachepermission) or [TopicPermission](#topicpermission), and the array can contain [up to ten](../../manage/limits) permission objects. A permission only grants access to the Momento data plane APIs (e.g. `get`, `set`, etc.). When an auth token is created with multiple permission objects, any matching permission will grant access. For example, if a single token is created with two permission objects:
+
+1. A permission object that allows ReadWrite access to all caches for the account 
+2. A permission object that allows ReadOnly access to cache `foo`
+
+In this case, the token will still allow use of data manipulation APIs (e.g. `set`, `delete`, `DictionarySetFields`, etc.) on cache `foo` because of the first permission.
+
+## Permission Objects
+
+These objects define the specific role with cache or topic information and then assigned to a [TokenScope](#tokenscope). 
+
+### CachePermission
+A component of a [TokenScope](#tokenscope) object that defines permissions for a cache.
+
+| Name            | Type                  | Description                                                                           |
+| --------------- |-----------------------|---------------------------------------------------------------------------------------|
+| role           | ReadOnly \| ReadWrite | The type of access granted by the permission.                                         |
+| cache          | `AllCaches` \| String | A permission can be restricted to a select cache by name or AllCaches built-in value. |
+
+For role, using `CacheRole.ReadOnly` allows access to all read data plane APIs on caches (e.g. `get`, `DictionaryGetField`, etc.) defined in the CacheSelector. Using `CacheRole.ReadWrite` allows access for all read and write data plane APIs on the caches defined in the CacheSelector. Custom roles are not supported.
+
+For cache, the value can be the built-in `AllCaches` or a string value containing the name of the cache this permission is for.
+
+### TopicPermission
+A component of a [TokenScope](#tokenscope) object that defines permissions for a token.
+
+| Name            | Type                              | Description                                                                          |
+| --------------- |-----------------------------------|--------------------------------------------------------------------------------------|
+| role           | SubscribeOnly \| PublishSubscribe | The type of access granted by the permission.               |
+| cache          | `AllCache` \| String              | A permission can be restricted to a select cache by name or AllCaches built-in value.              |
```

Comment:
It can also be an object that satisfies the CacheName interface ` { name: string }`. The string is a shorthand. We may add other types of selectors like Prefix in future.
Similar comment for `topic` which can be a TopicName

Reply:
Not saying no, but for simplicity's sake, why would anyone ever use the object when they could just use a simple string? My thought is if in the future we change things, then we could put that in. Thoughts?

Reply:
Because I am not sure we can expose this as a string in all SDKs - we may end up having CacheName/TopicName classes in other languages.

Reply:
This documentation is common to all SDKs so it should define the fields logically rather than use specific data types.

Reply:
That makes far more sense to me and was not clear before. will change!